beacon-chain/rpc/beacon/validators_stream.go

```diff
@@ -0,0 +1,379 @@
+package beacon
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/big"
+	"sort"
+	"sync"
+	"time"
+
+	cache "github.com/patrickmn/go-cache"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/feed"
+	statefeed "github.com/prysmaticlabs/prysm/beacon-chain/core/feed/state"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// eth1Deposit contains information about a deposit made on the Ethereum 1 chain.
+type eth1Deposit struct {
+	block *big.Int
+	data  *ethpb.Deposit_Data
+}
+
+// StreamValidatorsInfo returns a stream of information for given validators.
+// Validators are supplied dynamically by the client, and can be added, removed and reset at any time.
+func (bs *Server) StreamValidatorsInfo(stream ethpb.BeaconChain_StreamValidatorsInfoServer) error {
+	pubKeys := make([][]byte, 0)
+	pubKeysMutex := sync.RWMutex{}
+	stateChannel := make(chan *feed.Event, 1)
+	stateSub := bs.StateNotifier.StateFeed().Subscribe(stateChannel)
+	eth1Deposits := cache.New(time.Duration(params.BeaconConfig().SecondsPerSlot*params.BeaconConfig().SlotsPerEpoch)*time.Second, 12*time.Minute)
```

Comment:
maybe it will be better to name those vars and make prune time a multiple of epoch time

Reply:
Thanks; named and tidied up.